content/docs/04-ai-sdk-rsc/02-streaming-react-components.mdx

```diff
@@ -201,3 +201,50 @@ This page is first marked as a client component with the `"use client";` directi
 You can now allow the model to respond to your prompt with a React component. However, this example is limited to a static prompt that is set within your Server Action. You could make this example interactive by turning it into a chatbot.
 
 Learn how to stream React components with the Next.js App Router using `streamUI` with this [example](https://sdk.vercel.ai/examples/next-app/interface/route-components).
+
+## Advanced Usage
+In some cases, you might need to stream text or components within a tool call of a `streamUI` function. This can be achieved by nesting another `streamUI` call inside the tool. Here's an example of how you might structure this:
+
+```typescript
+const result = await streamUI({
+  model: openai('gpt-4'),
+  prompt: 'Create a data visualization dashboard',
+  text: ({ content }) => <div>{content}</div>,
+  tools: {
+    generateDashboard: {
+      description: 'Generate a data visualization dashboard based on user specifications',
+      parameters: z.object({
+        specifications: z.string()
+      }),
+      generate: async function* ({ specifications }) {
+        yield <LoadingComponent message="Generating dashboard..." />;
+        
+        const dashboardResult = await streamUI({
+          model: openai('gpt-4'),
+          prompt: `Create a dashboard with the following specifications: ${specifications}`,
+          text: ({ content, done, delta }) => {
+            // Handle streaming text updates here
+            return <DashboardGenerationProgress content={content} />;
+          },
+          tools: {
+            // Additional tools for dashboard generation can be defined here
+          }
+        });
+
+        return <DashboardComponent>{dashboardResult.value}</DashboardComponent>;
+      },
+    },
+  },
```

Comment:
I'm a bit torn about this pattern and whether we want to promote it, since it leads to complications with creating a consistent chat history, especially with parallel tool calling.

I plan to revisit this once I'm done with the streamUI refactoring.